File: src/main/java/net/insprill/cjm/hooks/EssentialsHook.java
package net.insprill.cjm.hooks;


import com.earth2me.essentials.Essentials;
import net.ess3.api.events.VanishStatusChangeEvent;
import net.insprill.cjm.handlers.VanishHandler;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;

public class EssentialsHook implements Listener {

    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    public void onVanish(VanishStatusChangeEvent e) {
        VanishHandler.handleVanishToggle(e.getAffected().getBase(), e.getValue());
    }

    /**
     * Checks if a player is jailed.
     *
     * @param player Player to check.
     * @return True if the player is in Jail, false otherwise.
     */
    public static boolean isJailed(Player player) {
        return Essentials.getPlugin(Essentials.class).getUser(player).isJailed();
    }

}

File: src/main/kotlin/net/insprill/cjm/compatibility/hook/EssentialsHook.kt
package net.insprill.cjm.compatibility.hook

import com.earth2me.essentials.Essentials
import net.ess3.api.events.VanishStatusChangeEvent
import net.insprill.cjm.CustomJoinMessages
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener

class EssentialsHook(private val plugin: CustomJoinMessages) : Listener {

    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    fun onVanish(e: VanishStatusChangeEvent) {
        plugin.vanish.handleToggle(e.affected.base, e.value)
    }

}
